### DISCUSSION TITLE
Quarkus Redis connection issues

### DISCUSSION THREAD

**Discussion by krezchikov42** [quarkusio/quarkus#40527] [Q&A]
So I have been seeing these errors lately. 
`"io.vertx.core.impl.NoStackTraceThrowable: Failed to connect to all nodes of the cluster"` 
I have seen these mentioned in issues like this https://github.com/quarkusio/quarkus/issues/37348 and https://github.com/quarkusio/quarkus/issues/37041
However, I'm not sure if using the `ReactiveRedisDataSource` I am allowing for the connection to reconnect on error. I know in [Vert.x](https://vertx.io/docs/vertx-redis-client/java/#_implementing_reconnect_on_error)  you have to manually implement the reconnect on errror. How would I do this when I am using the `ReactiveRedisDataSource`? 
I don't see any information about reconnecting on error in the quarkus redis [documentation](https://quarkus.io/guides/redis-reference#use-the-replication-mode)

[a comment by an automated account is not shown]

**Comment by Ladicek** [comment]
There is a straightforward reconnect on error functionality in the Vert.x Redis client (contrary to the documentation, perhaps that should be updated) in the form of connection pooling. When a connection enters an error state, it is removed from the pool and a new one is created.

Could you use a newer Quarkus version? It includes a newer Vert.x Redis client, which includes more descriptive error messages.
